tree/dataframe/inc/ROOT/RDF/RInterface.hxx

```diff
@@ -645,10 +645,13 @@ public:
                 << RDFInternal::PrettyPrintAddr(&upcastInterface) << ")->Cache<";
 
       const auto &customCols = fCustomColumns.GetNames();
+      const auto &aliasMap = fLoopManager->GetAliasMap();
       for (auto &c : columnList) {
-         const auto isCustom = std::find(customCols.begin(), customCols.end(), c) != customCols.end();
-         const auto customColID = isCustom ? fCustomColumns.GetColumns().at(c)->GetID() : 0;
-         cacheCall << RDFInternal::ColumnName2ColumnTypeName(c, nsID, tree, fDataSource, isCustom,
+         const auto isAlias = aliasMap.find(c) != aliasMap.end();
+         const auto resolvedName = isAlias ? aliasMap.at(c) : c;
+         const auto isCustom = std::find(customCols.begin(), customCols.end(), resolvedName) != customCols.end();
+         const auto customColID = isCustom ? fCustomColumns.GetColumns().at(resolvedName)->GetID() : 0;
+         cacheCall << RDFInternal::ColumnName2ColumnTypeName(resolvedName, nsID, tree, fDataSource, isCustom,
```

Comment:
I think the fix should be just:

```patch
diff --git a/tree/dataframe/inc/ROOT/RDF/RInterface.hxx b/tree/dataframe/inc/ROOT/RDF/RInterface.hxx
index f0284e26bf..07097bce5e 100644
--- a/tree/dataframe/inc/ROOT/RDF/RInterface.hxx
+++ b/tree/dataframe/inc/ROOT/RDF/RInterface.hxx
@@ -645,7 +645,8 @@ public:
                 << RDFInternal::PrettyPrintAddr(&upcastInterface) << ")->Cache<";
 
       const auto &customCols = fCustomColumns.GetNames();
-      for (auto &c : columnList) {
+      const auto validColumnNames = GetValidatedColumnNames(columnList.size(), columnList);
+      for (auto &c : validColumnNames) {
          const auto isCustom = std::find(customCols.begin(), customCols.end(), c) != customCols.end();
          const auto customColID = isCustom ? fCustomColumns.GetColumns().at(c)->GetID() : 0;
          cacheCall << RDFInternal::ColumnName2ColumnTypeName(c, nsID, tree, fDataSource, isCustom,
```

This keeps Cache logic in line with Snapshot, and does not require to deal with aliases manually here. Seems to work on my workstation.

Reply:
But then we throw away the alias names?

Reply:
Ah ok, I think I understood now the mechanics of that thingy. So I agree that this should be fine since the `GetValidatedColumnNames` replaces the aliases by the actual column names.

Reply:
Though it's not clear to me whether it's safe since the list of validated names could differ (in length) from the original one and the jitted code won't fit the user's request.

Reply:
> the list of validated names could differ (in length) from the original one

I don't think it can..?

Reply:
The function is weird since you can give the `nColumns` argument, which indicates this.

Reply:
In general it can return the original column list plus some default columns. Not here, because we call it specifying that we want as many columns as already are in the columnList